conda_forge_tick/auto_tick.py

```diff
@@ -554,67 +558,64 @@ def run(
         logger.warning("Skipping migration due to solvability check failure")
         return False, False
 
-    # This is needed because we want to migrate to the new backend step-by-step
-    repo: github3.repos.Repository | None = github3_client().repository(
-        context.git_repo_owner, context.git_repo_name
-    )
-
-    assert repo is not None
-
-    feedstock_dir = str(context.local_clone_dir.resolve())
-
-    # TODO: Better annotation here
-    pr_json: typing.Union[MutableMapping, None, bool]
+    pr_data: PullRequestData | PullRequestInfoSpecial | None
+    """
+    The PR data for the PR that was created. The contents of this variable will be stored in the bot's database.
+    None means: We don't update the PR data.
+    """
     if (
         isinstance(migrator, MigrationYaml)
         and not rerender_info.nontrivial_changes
         and context.attrs["name"] != "conda-forge-pinning"
     ):
         # spoof this so it looks like the package is done
-        pr_json = {
-            "state": "closed",
-            "merged_at": "never issued",
-            "id": str(uuid4()),
-        }
+        pr_data = get_spoofed_closed_pr_info()
     else:
-        # push up
+        # push and PR
+        git_backend.push_to_repository(
```

Comment:
Why are we no longer catching errors on the repo push?

Reply:
In the old code, the try-except block looks like this:

https://github.com/regro/cf-scripts/blob/621910b1438b5ed2f3510fc72c7f692ac4e2ed5a/conda_forge_tick/auto_tick.py#L581-L601

In `push_repo`, the only place where a `github3.GitHubError` can be raised is:

https://github.com/regro/cf-scripts/blob/621910b1438b5ed2f3510fc72c7f692ac4e2ed5a/conda_forge_tick/git_utils.py#L1115

`push_to_repository` only executes `git push`, the PR is created in `create_pull_request` below, which is protected by a try-else block catching duplicate PRs. To my understanding, this should all be equivalent to the old code.

Reply:
ah got it. SGTM and thanks for checking!